### PR TITLE
android-ndk: define `tools.build:compiler_executables` config

### PR DESCRIPTION
**Pull request by SpaceIm**
define new `tools.build:compiler_executables` config introduced in conan 1.55.0, see https://github.com/conan-io/conan/pull/12556

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/android-ndk/all/conanfile.py [hunk 1]
```diff
@@ -281,8 +281,13 @@ def package_info(self):
         # when `tools.android:ndk_path` is provided, so it MUST NOT be manually injected here to `tools.cmake.cmaketoolchain:user_toolchain` conf_info
         self.conf_info.define("tools.android:ndk_path", os.path.join(self.package_folder, "bin"))
 
-        self.buildenv_info.define_path("CC", self._define_tool_var("CC", "clang"))
-        self.buildenv_info.define_path("CXX", self._define_tool_var("CXX", "clang++"))
+        compiler_executables = {
+            "cc": self._define_tool_var("CC", "clang"),
+            "cxx": self._define_tool_var("CXX", "clang++"),
```
Comment (franramirez688) [inline recipes/android-ndk/all/conanfile.py:286]:
```suggestion
            "cpp": self._define_tool_var("CXX", "clang++"),
```

File: recipes/android-ndk/all/conanfile.py [hunk 2]
```diff
@@ -281,8 +281,13 @@ def package_info(self):
...
+        }
+        self.conf_info.define("tools.build:compiler_executables", compiler_executables)
+        self.buildenv_info.define_path("CC", compiler_executables.get("cc"))
+        self.buildenv_info.define_path("CXX", compiler_executables.get("cxx"))
```
Comment (franramirez688) [inline recipes/android-ndk/all/conanfile.py:290]:
```suggestion
        self.buildenv_info.define_path("CXX", compiler_executables.get("cpp"))
```

File: recipes/android-ndk/all/conanfile.py [hunk 3]
```diff
@@ -332,8 +337,8 @@ def package_info(self):
             cmake_wrapper = os.path.join(self.package_folder, "bin", cmake_wrapper)
             self.env_info.CONAN_CMAKE_PROGRAM = cmake_wrapper
             self.env_info.CONAN_CMAKE_TOOLCHAIN_FILE = os.path.join(self.package_folder, "bin", "build", "cmake", "android.toolchain.cmake")
-            self.env_info.CC = self._define_tool_var("CC", "clang")
-            self.env_info.CXX = self._define_tool_var("CXX", "clang++")
+            self.env_info.CC = compiler_executables.get("cc")
+            self.env_info.CXX = compiler_executables.get("cxx")
```
Comment (franramirez688) [inline recipes/android-ndk/all/conanfile.py:341]:
```suggestion
            self.env_info.CXX = compiler_executables.get("cpp")
```

File: recipes/android-ndk/all/conanfile.py [hunk 4]
```diff
@@ -281,8 +281,13 @@ def package_info(self):
         # when `tools.android:ndk_path` is provided, so it MUST NOT be manually injected here to `tools.cmake.cmaketoolchain:user_toolchain` conf_info
         self.conf_info.define("tools.android:ndk_path", os.path.join(self.package_folder, "bin"))
 
-        self.buildenv_info.define_path("CC", self._define_tool_var("CC", "clang"))
-        self.buildenv_info.define_path("CXX", self._define_tool_var("CXX", "clang++"))
+        compiler_executables = {
+            "cc": self._define_tool_var("CC", "clang"),
```
Comment (franramirez688) [inline recipes/android-ndk/all/conanfile.py:285]:
```suggestion
            "c": self._define_tool_var("CC", "clang"),
```

File: recipes/android-ndk/all/conanfile.py [hunk 5]
```diff
@@ -281,8 +281,13 @@ def package_info(self):
...
+            "cxx": self._define_tool_var("CXX", "clang++"),
+        }
+        self.conf_info.define("tools.build:compiler_executables", compiler_executables)
+        self.buildenv_info.define_path("CC", compiler_executables.get("cc"))
```
Comment (franramirez688) [inline recipes/android-ndk/all/conanfile.py:289]:
```suggestion
        self.buildenv_info.define_path("CC", compiler_executables.get("c"))
```

File: recipes/android-ndk/all/conanfile.py [hunk 6]
```diff
@@ -332,8 +337,8 @@ def package_info(self):
             cmake_wrapper = os.path.join(self.package_folder, "bin", cmake_wrapper)
             self.env_info.CONAN_CMAKE_PROGRAM = cmake_wrapper
             self.env_info.CONAN_CMAKE_TOOLCHAIN_FILE = os.path.join(self.package_folder, "bin", "build", "cmake", "android.toolchain.cmake")
-            self.env_info.CC = self._define_tool_var("CC", "clang")
-            self.env_info.CXX = self._define_tool_var("CXX", "clang++")
+            self.env_info.CC = compiler_executables.get("cc")
```
Comment (franramirez688) [inline recipes/android-ndk/all/conanfile.py:340]:
```suggestion
            self.env_info.CC = compiler_executables.get("c")
```

**Review by franramirez688 — COMMENTED** [review]
Hi @SpaceIm - Thanks for the contribution!

The documentation for 1.55 version is not deployed yet, but the keys that are going to be mapped as `{'c', 'cpp', 'cuda', 'objc', 'objcpp', 'rc', 'fortran', 'asm', 'hip', 'ispc'}` for now. I say "for now" because I guess that we'll add more key-value's in the future.

File: recipes/android-ndk/all/conanfile.py [hunk 7]
```diff
@@ -281,8 +281,13 @@ def package_info(self):
         # when `tools.android:ndk_path` is provided, so it MUST NOT be manually injected here to `tools.cmake.cmaketoolchain:user_toolchain` conf_info
         self.conf_info.define("tools.android:ndk_path", os.path.join(self.package_folder, "bin"))
 
-        self.buildenv_info.define_path("CC", self._define_tool_var("CC", "clang"))
-        self.buildenv_info.define_path("CXX", self._define_tool_var("CXX", "clang++"))
+        compiler_executables = {
+            "c": self._define_tool_var("CC", "clang"),
+            "cpp": self._define_tool_var("CXX", "clang++"),
+        }
```
Comment (SpaceIm) [inline recipes/android-ndk/all/conanfile.py:287]: @franramirez688 should it be replaced by:

```suggestion
        compiler_executables = self.conf.get("tools.build:compiler_executables", default={}, check_type=str)
        compiler_executables["c"] = self._define_tool_var("CC", "clang")
        compiler_executables["cpp"] = self._define_tool_var("CXX", "clang++")
```

Comment (franramirez688) [reply recipes/android-ndk/all/conanfile.py:287]: Good point! 👏 

Anyway, I recall that we could simplify these lines:

```python
            compiler_executables = {
	            "c": self._define_tool_var("CC", "clang"),
	            "cpp": self._define_tool_var("CXX", "clang++"),
	        }
            self.conf_info.update("tools.build:compiler_executables", compiler_executables)
```
If I'm not wrong, this should do the trick.

**Review by uilianries — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**